### PR TITLE
Migrate helpers, syncoptions and tests for TaxCategory Sync

### PR DESCRIPTION
**Pull request by salander85**
JIRA: https://commercetools.atlassian.net/browse/DEVX-157

### CODE REVIEW

File: src/main/java/com/commercetools/sync/sdk2/taxcategories/helpers/TaxCategoryBatchValidator.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+            "statusCode": 400,
+            "message": "A duplicate value '{\"country\":\"DE\"}' exists for field 'country'.",
+            "errors": [
+                {
+                    "code": "DuplicateField",
+                    ....
+            ]
+        }
+    */
+    Map<String, Map<String, Long>> map =
```
Comment (lojzatran) [inline src/main/java/com/commercetools/sync/sdk2/taxcategories/helpers/TaxCategoryBatchValidator.java:109]:
```suggestion
    final Map<String, Map<String, Long>> map =
```

File: src/main/java/com/commercetools/sync/sdk2/taxcategories/helpers/TaxCategoryBatchValidator.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+            .collect(
+                Collectors.groupingBy(
+                    draft -> Objects.toString(draft.getCountry(), ""),
+                    Collectors.groupingBy(
+                        draft -> Objects.toString(draft.getState(), ""), Collectors.counting())));
+
+    for (Map.Entry<String, Map<String, Long>> countryEntry : map.entrySet()) {
+      for (Map.Entry<String, Long> stateEntry : countryEntry.getValue().entrySet()) {
+        if (stateEntry.getValue() > 1L) {
+          String errorMessage =
```
Comment (lojzatran) [inline src/main/java/com/commercetools/sync/sdk2/taxcategories/helpers/TaxCategoryBatchValidator.java:120]:
```suggestion
          final String errorMessage =
```

File: src/test/java/com/commercetools/sync/sdk2/taxcategories/helpers/TaxCategoryBatchValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+        getValidDrafts(Collections.singletonList(taxCategoryDraft));
+
+    assertThat(errorCallBackMessages).hasSize(1);
+    assertThat(errorCallBackMessages.get(0))
+        .isEqualTo(format(TAX_CATEGORY_DRAFT_KEY_NOT_SET, taxCategoryDraft.getName()));
+    assertThat(validDrafts).isEmpty();
+  }
+
+  @Test
+  void validateAndCollectReferencedKeys_WithMixOfValidAndInvalidDrafts_ShouldValidateCorrectly() {
```
Comment (lojzatran) [inline src/test/java/com/commercetools/sync/sdk2/taxcategories/helpers/TaxCategoryBatchValidatorTest.java:90]: Please add validation tests for:
1. different country, same state
2. country is null

**Review by lojzatran — CHANGES_REQUESTED** [review]

File: src/main/java/com/commercetools/sync/sdk2/taxcategories/helpers/TaxCategoryBatchValidator.java (hunk 2 above)
Comment (salander85) [reply src/main/java/com/commercetools/sync/sdk2/taxcategories/helpers/TaxCategoryBatchValidator.java]: [12c545b](https://github.com/commercetools/commercetools-sync-java/pull/1039/commits/12c545bd408f2c6f7cbaa2ae91a137de32588666)

File: src/test/java/com/commercetools/sync/sdk2/taxcategories/helpers/TaxCategoryBatchValidatorTest.java (hunk 3 above)
Comment (salander85) [reply src/test/java/com/commercetools/sync/sdk2/taxcategories/helpers/TaxCategoryBatchValidatorTest.java:90]: [52ea8d5](https://github.com/commercetools/commercetools-sync-java/pull/1039/commits/52ea8d5f84c71032f5d7e6fdf7957456a8222c1f)

**Pull request merged**